### ISSUE TITLE
trained model error

### ISSUE THREAD

**Issue by gjtjx** [Zehaos/MobileNet#15]
I have trained mobilenet on my own dataset, the training run 700000 iters,loss is 2e-3, the image in tensor board is right, when training a warring

WARNING:tensorflow:Error encountered when serializing LAYER_NAME_UIDS.
Type is unsupported, or the types of the items don't match field type in CollectionDef.

but when I eval_image_classifier test, the Accuracy is very low,
2017-05-26 19:38:54.825458: I tensorflow/core/kernels/logging_ops.cc:79] eval/Accuracy[0.13553691]
2017-05-26 19:38:54.825467: I tensorflow/core/kernels/logging_ops.cc:79] eval/Recall_5[0.26291946]
and i write the eval as 
network_fn = nets_factory.get_network_fn(
        'mobilenet',
        num_classes=(59532),
        is_training=False,
        width_multiplier=1.0)
logits, end_points = network_fn(input_tensor)
...
predict_values, logit_values = sess.run([end_points['squeeze'], logits], feed_dict={input_tensor: img})

it says MobileNet/Conv1/Weight2 is not initialized

ii anything wrong?

**Comment by Zehaos** [comment]
I assume that you have make the tfrecord correctly. (Both test set and train set)

And you said that you used 'eval_image_classifier.py' to do the evaluation and got a low accuracy. 

Then you write a new evaluation script??? And it come up with 'MobileNet/Conv1/Weight2 is not initialized'??? Right?

**Comment by Zehaos** [comment]
To figure out whether there is any problem with the evaluation procedure, I think you could use the training set to do the evaluation. If the accuracy is high (as the training loss is low enough), it means there is no problem with the evaluation program.

**Comment by gjtjx** [comment]
Sorry，I have check the test img , the bbox is wrong，
now the test Accuracy is normal
2017-05-26 20:05:00.298230: I tensorflow/core/kernels/logging_ops.cc:79] eval/Accuracy[0.99077183]
2017-05-26 20:05:00.298274: I tensorflow/core/kernels/logging_ops.cc:79] eval/Recall_5[0.99731543624161079]

but how can i extract the avg_pool_15 feature? Thank you

**Comment by Zehaos** [comment]
Glad to hear that~

To extract the feature, you can use 
```
end_point['avg_pool_15']
```

**Comment by Zehaos** [comment]
Are you applying MobileNet to face recognition task? The result seems quite well.

**Comment by gjtjx** [comment]
I am on person re-id task
evaluate step as 
slim.evaluation.evaluate_once(
        master=FLAGS.master,
        checkpoint_path=checkpoint_path,
        logdir=FLAGS.eval_dir,
        num_evals=num_batches,
        eval_op=list(names_to_updates.values()),
        variables_to_restore=variables_to_restore)
how to get end_point by this function?

**Comment by Zehaos** [comment]
Cool~ 
It is a high level API. You have to hack into it or write a new script to extract the feature. It is not so difficult to write one.
